### PR TITLE
WIP:File storage management

### PR DESCRIPTION
**Pull request by Dromadon**

### CODE REVIEW

File: app/hallebarde/infrastructure/file_repository.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+from typing import List
+
+import boto3
+import hallebarde.config
+
+BUCKET_NAME = f'hallebarde-storage-{hallebarde.config.ENVIRONMENT}'
+
+
+def get_files_list(identifier: str) -> List:
```
Comment (Dromadon) [inline app/hallebarde/infrastructure/file_repository.py]: Set get_files_list private and add a get_file(identifier) method which returns only the last added file

File: app/hallebarde/infrastructure/exchange_repository.py [hunk 2]
```diff
@@ -31,6 +31,16 @@ def delete(identifier: str) -> None:
     table.delete_item(Key={'identifier': identifier})
 
 
+def get_identifier_from_token(upload_token: str = None, download_token: str = None) -> Optional[str]:
```
Comment (Dromadon) [inline app/hallebarde/infrastructure/exchange_repository.py]: up/dl_token are Optional[str]

File: app/hallebarde/infrastructure/exchange_repository.py [hunk 3]
```diff
@@ -31,6 +31,16 @@ def delete(identifier: str) -> None:
...
+    table = _get_dynamodb_table()
+    if upload_token:
+        response = table.scan(FilterExpression=Attr('upload_token').eq(upload_token))
```
Comment (Dromadon) [inline app/hallebarde/infrastructure/exchange_repository.py]: Extraire le table.scan en dessous, juste set Attr = … token = …

File: app/tests/infrastructure/test_exchange_repository.py [hunk 4]
```diff
@@ -64,3 +64,34 @@ def test_delete_should_delete_indicated_exchange(self, mock_get_table, an_exchan
...
+        # When
+        identifier_from_upload_token = exchange_repository.get_identifier_from_token(upload_token=an_exchange.upload_token)
+        identifier_from_download_token = exchange_repository.get_identifier_from_token(download_token=an_exchange.download_token)
+
+        # Then
+        assert identifier_from_upload_token == an_exchange.identifier
+        assert identifier_from_download_token == an_exchange.identifier
+
+    @patch('hallebarde.infrastructure.exchange_repository._get_dynamodb_table')
+    def test_get_identifier_from_token_should_return_none_if_exchange_does_not_exist(self, mock_get_table,
```
Comment (Dromadon) [inline app/tests/infrastructure/test_exchange_repository.py]: Add a NoneSafeness test -> what if up and dl token are None ? should return None then
